lib/msal-browser/src/broker/client/BrokerClientApplication.ts

```diff
@@ -136,13 +165,23 @@ export class BrokerClientApplication extends ClientApplication {
         if (validMessage) {
             this.logger.verbose(`Broker auth request validated: ${validMessage}`);
 
-            // TODO: Calculate request thumbprint
-            const brokerResult = await this.cachedBrokerResponse;
-            if (brokerResult && brokerResult.tokensToCache) {
-                // TODO: Replace with in-memory cache lookup
-                this.cachedBrokerResponse = undefined;
+            if (this.currentBrokerRedirectResponse) {
+                await this.currentBrokerRedirectResponse;
+            }
+
+            const reqThumbprint: RequestThumbprint = {
+                authority: validMessage.request.authority,
+                clientId: validMessage.embeddedClientId,
+                scopes: validMessage.request.scopes
+            };
+            const responseThumbprint = `${validMessage.embeddedAppOrigin}.${this.browserCrypto.base64Encode(JSON.stringify(reqThumbprint))}`;
+            const responseCacheKey = `${MemoryCacheKeys.BROKER_RESPONSE}.${responseThumbprint}`;
```

Comment:
Can you factor these into helper functions? Would help readability.

Reply:
done.